File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        def sumDig(num: int) -> int:
            res = 0
            while num > 0:
                res += num % 10
                num //= 10
            return res
        for i in range(len(nums)):
            if i == sumDig(nums[i]):
                return i
        return -1

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.cpp
class Solution {
public:
    int smallestIndex(vector<int>& nums) {
        for(int i = 0; i < nums.size(); i++)
        {
            if(i == sumDig(nums[i]))
            {
                return i;
            }
        }
        return -1;
    }
private:
    int sumDig(int num)
    {
        int res = 0;
        while(num > 0)
        {
            res += num % 10;
            num /= 10;
        }
        return res;
    }
};